cmd/account/account-assign.go

```diff
@@ -0,0 +1,167 @@
+
+	return accountAssignCmd
+}
+
+type accountAssignOptions struct {
+	username     string
+	payerAccount string
+	output       string
+
+	flags      *genericclioptions.ConfigFlags
+	printFlags *printer.PrintFlags
+	genericclioptions.IOStreams
+	kubeCli client.Client
+}
+
+func newAccountAssignOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *accountAssignOptions {
+	return &accountAssignOptions{
+		flags:      flags,
+		printFlags: printer.NewPrintFlags(),
+		IOStreams:  streams,
+	}
+}
+
+func (o *accountAssignOptions) complete(cmd *cobra.Command, _ []string) error {
+	if o.username == "" {
+		return cmdutil.UsageErrorf(cmd, "Red Hat username was not provided")
+	}
+
+	var err error
+	o.kubeCli, err = k8s.NewClient(o.flags)
+	if err != nil {
+		fmt.Println(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (o *accountAssignOptions) run() error {
+
+	var (
+		accountAssignID string
+		destinationOu   string
+		nonDefaultPayer string
+		rootId          string
+		defaultPayer    = "osd-staging-2"
+		claimTag        = "claimed"
+		claimTagValue   = "true"
+	)
+
+	o.payerAccount = defaultPayer
+	rootId = profileRootId2
+	if o.username != "" && o.payerAccount != "" {
+		nonDefaultPayer = o.payerAccount
+		rootId = profileRootId1
+	}
+	//Instantiate aws client
+	awsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		fmt.Println(err.Error())
+		return err
+	}
+	//List accounts that are not in any OU
+	input := &organizations.ListAccountsInput{}
+	accounts, err := awsClient.ListAccounts(input)
+	if err != nil {
+		fmt.Println(err.Error())
+		return err
+	}
+	//Create input for tagging
+	inputTag := &organizations.TagResourceInput{
+		ResourceId: aws.String(accountAssignID),
+		Tags: []*organizations.Tag{
+			{
+				Key:   aws.String("owner"),
+				Value: aws.String(o.username),
+			},
+			{
+				Key:   aws.String(claimTag),
+				Value: aws.String(claimTagValue),
+			},
+		},
+	}
+
+	//Loop through the list of accounts and get ID
+	for _, a := range accounts.Accounts {
+		accountAssignID = *a.Id
+		//Tag account
+		_, err = awsClient.TagResource(inputTag)
+		if err != nil {
+			fmt.Println(err.Error())
+			return err
+		}
+		break
+	}
+
+	//Move account to developers OU
+	inputMove := &organizations.MoveAccountInput{
+		AccountId:           aws.String(accountAssignID),
+		DestinationParentId: aws.String(destinationOu),
+		SourceParentId:      aws.String(rootId),
+	}
+
+	if o.payerAccount == defaultPayer {
+		destinationOu = defaultBYOCEnv
+		rootId = profileRootId2
+		_, err = awsClient.MoveAccount(inputMove)
+		if err != nil {
+			fmt.Println(err.Error())
+			return err
+		}
+	} else if o.payerAccount == nonDefaultPayer {
+		destinationOu = defaultNonBYOCEnv
+		rootId = profileRootId1
+		_, err = awsClient.MoveAccount(inputMove)
+		if err != nil {
+			fmt.Println(err.Error())
+			return err
+		}
+	}
```

Comment:
I'm not sure if this is going to work correctly, although it might because of the pointers, but it's not very clear.  It might be better to do something like:

```
inputMove := organizations.MoveAccountInput

if default payer account {
  fill in inputMove with defaultPayeraccount stuff
} else {
  fill in inputMove with non-default stuff
}

_, err = awsClient.MoveAccount(inputMove)
if err != nil {
  fmt.Println(err.Error())
  return err
}
return nil